src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,68 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a movie session by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        String query = "FROM MovieSession ms left join fetch ms.movie m where m.id = :id";
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> movieSessionQuery = session.createQuery(query, MovieSession.class);
+            movieSessionQuery.setParameter("id", movieId);
+            List<MovieSession> allMovieSessions = movieSessionQuery.getResultList();
+            return findSessionsByDate(allMovieSessions, date);
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get movie sessions in date: "
+                    + date + " by movie id: " + movieId, e);
+        }
+    }
+
+    private List<MovieSession> findSessionsByDate(List<MovieSession> sessions, LocalDate date) {
+        return sessions.stream()
+                .filter(movieSession -> movieSession.getShowTime().toLocalDate().equals(date))
+                .collect(Collectors.toList());
+    }
```

Comment:
lets create query to get movie sessions between date at the beginning of the day and end 
```
date.atTime(LocalTime.MAX)
date.atTime(LocalTime.MIN)
```